app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala

```diff
@@ -136,7 +129,37 @@ class MovementsController @Inject() (
       )
   }
 
-  def updateMovement(movementId: MovementId, triggerId: Option[MessageId] = None): Action[Source[ByteString, _]] =
+  def updateMovement(movementId: MovementId, triggerId: Option[MessageId] = None) =
+    contentTypeRoute {
+      case Some(_) => updateMovementSmallMessage(movementId, triggerId)
+      case None    => updateMovementLargeMessage(movementId, triggerId)
+    }
+
+  def updateMovementLargeMessage(movementId: MovementId, triggerId: Option[MessageId] = None): Action[AnyContent] = Action.async(parse.anyContent) {
+    implicit request =>
+      (for {
+        messageType    <- extract(request.headers).asPresentation
+        objectStoreURL <- extractObjectStoreURI(request.headers).asPresentation
+        objectStoreFilePath = if (objectStoreURL.contains("/")) objectStoreURL.split("/", 2).apply(1) else objectStoreURL
+        sourceFile  <- objectStoreService.getObjectStoreFile(objectStoreFilePath).asPresentation
+        messageData <- messagesXmlParsingService.extractMessageData(sourceFile, messageType).asPresentation
+        received = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC)
+        message = messageFactory
+          .createLargeMessage(
+            messageType,
+            messageData.generationDate,
+            received,
+            triggerId,
+            new URI(objectStoreURL)
+          )
+        _ <- repo.updateMessages(movementId, message, messageData.mrn, received).asPresentation
+      } yield message.id).fold[Result](
+        baseError => Status(baseError.code.statusCode)(Json.toJson(baseError)),
+        id => Ok(Json.toJson(UpdateMovementResponse(id)))
+      )
+  }
+
+  private def updateMovementSmallMessage(movementId: MovementId, triggerId: Option[MessageId] = None): Action[Source[ByteString, _]] =
```

Comment:
Might be good to create a Small messages and Large messages controller, but not required in this PR